lib/class-wp-theme-json.php

```diff
@@ -1197,6 +1194,19 @@ private static function remove_insecure_styles( $input ) {
 		return $output;
 	}
 
+	/**
+	 * Checks that a declaration provided by the user is safe.
+	 *
+	 * @param string $property_name Property name in a CSS declaration, i.e. the `color` in `color: red`.
+	 * @param string $property_value Value in a CSS declaration, i.e. the `red` in `color: red`.
+	 * @return boolean
+	 */
+	private static function validate_style_declaration( $property_name, $property_value ) {
+		$style_to_validate = $property_name . ': ' . $property_value;
+		$filtered          = esc_html( safecss_filter_attr( $style_to_validate ) );
+		return ! empty( trim( $filtered ) );
```

Comment:
This is a better check, thanks! By reading the `safecss_filter_attr` code I've found other cases in which the current code could have failed:

- the input string has some null value (null, backspace)
- the input string has new line characters (`\t`, `\r`, `\n`)
- the input string has leading or trailing whitespace

* note that "input string" is `property: value`